src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java

```diff
@@ -425,6 +430,19 @@ protected void checkImageWithAutoPull(ServiceHub hub, String image, String regis
         }
     }
 
+    private synchronized Set<String> getPreviouslyPulledImageCache() {
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        Set<String> cache = (Set) session.getUserProperties().get(CONTEXT_KEY_PREVIOUSLY_PULLED);
+        if (cache == null) {
+            System.out.println("creating new context");
+
+            cache = Sets.newConcurrentHashSet();
```

Comment:
I fixed it by introducing an `ImagePullCache` which is serializable (via JSONObject). Seems to work now with the session userProperties (it would already work with the executionProperties() but that's deprecated and putting sth else than a String as a Property value feels wrong anyway.